### ISSUE TITLE
Restoring checkpoint in next training run

### ISSUE THREAD

**Issue by KeirSimmons** [openai/baselines#106]
I see that (in simple.py) for DQN the model is saved at the end, and this model can then be loaded for playing the game with the learned parameters, but is it at all possible with the current framework to load the model and continue training from where it last saved?

**Comment by Sohojoe** [comment]
yes, but not out of the box - there are a couple of things you need to do:

a) make a minor update to simple.py 
b) tensorflow throws an error when restarting within the same program

here is a gist that shows how: https://gist.github.com/Sohojoe/25fe35364eba0ed0d3fb70b1e6068dcb

**Comment by kaihuchen** [comment]
@sohojoe   Thank you for sharing your code, which is exactly what I needed! However, for some reasons when I tried to restore a previously saved DQN model in order to continue with training it, for some reasons it appears to be always doing the restored training from scratch, and I can't seem to find any obvious errors. 

More specifically, I am using Baselines deepq for training a virtual car to navigate all by itself (demo video available here: https://www.youtube.com/watch?v=InrQgdU8rQs&feature=youtu.be ), which works well and the car learns to navigate around with good skill.  I used your code to initiate a training session to create a saved pickle file, then later restart another session to restore from this pickle file. I can see the pickle file being created, loaded, and a temp directory created during the restoration process, but for whatever reason the previously trained navigation skill is no longer there after restoration, thus it appears to me that the trained model is not being restored properly somewhere along the road.

Wonder if you have any insight you can share that will help me debug this problem? Thanks!

**Comment by llach** [comment]
Will there be any more work done for this feature? Or is there a technical issue that training cannot be continued from an already saved pickle?

Within the HER source code, a [comment](https://github.com/openai/baselines/blob/master/baselines/her/ddpg.py#L315) can be found that points out this problem but does not give any information to why that is. 

Having this feature would be great so we could continue training after it got interrupted somehow.

**Comment by pzhokhov** [comment]
@llach could you describe in more detail what issue are you facing? This OP's issue should have been already fixed (and I forgot to close this timely), but I'll leave this open if you have similar problems.

**Comment by llach** [comment]
We're currently evaluating the HER algorithm. Some issues with our filesystem cause the training script to abort sometimes when trying to write the log files or pickles to disk. 
If we'd be able to continue training from the last saved pickle, the time already invested in training wouldn't be wasted.
